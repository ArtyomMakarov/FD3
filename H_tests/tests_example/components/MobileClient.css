.MobileClient .cell .inputEdit, .MobileClient .cell .inputDelete
{
    width: 118px;
    padding: 5px;
}
