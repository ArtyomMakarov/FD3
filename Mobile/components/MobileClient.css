.MobileClient .cell .input
{
    width: 118px;
    padding: 5px;
}