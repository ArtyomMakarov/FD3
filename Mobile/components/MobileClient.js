import React from 'react';
import PropTypes from 'prop-types';

import './MobileClient.css';
import {voteEvents} from './events';

class MobileClient extends React.PureComponent {

    static propTypes = {
        info:PropTypes.shape({
            id: PropTypes.number.isRequired,
            fio: PropTypes.shape({
                last_name:PropTypes.string.isRequired,
                name:PropTypes.string.isRequired,
                middle_name:PropTypes.string.isRequired
            }),
            balance: PropTypes.number.isRequired,
            status: PropTypes.string.isRequired
        }),
        mode: PropTypes.number.isRequired
    };

    state = {
        info: this.props.info
    };

    componentWillReceiveProps = (newProps) => {
        this.setState({info:newProps.info});
    };

    deleteClient = () => {
        voteEvents.emit("EDeleteClient",this.props.info.id);
    };

    editClient = () => {
        voteEvents.emit("EEditClient",this.props.info.id);
    };

    render() {

        console.log("MobileClient id="+this.state.info.id+" render");

        return (
            <tr className='MobileClient'>
                <td className='cell'>{this.props.info.fio.last_name}</td>
                <td className='cell'>{this.props.info.fio.name}</td>
                <td className='cell'>{this.props.info.fio.middle_name}</td>
                <td className='cell'>{this.props.info.balance}</td>
                <td className='cell' style={{background: this.props.info.balance>0?"green":"red"}}>{this.props.info.status}</td>
                <td className='cell'>
                    <input type='button' className="input" value='Редактировать' onClick={this.editClient} disabled={this.props.mode==2}/>
                </td>
                <td className='cell'>
                    <input type='button' className="input" value='Удалить' onClick={this.deleteClient} disabled={this.props.mode>0}/>
                </td>
            </tr>
        );

    }

}

export default MobileClient;