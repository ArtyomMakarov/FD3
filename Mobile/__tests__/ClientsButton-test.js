"use strict";

import React from 'react';
import renderer from 'react-test-renderer';

import MobileClient from '../components/MobileClient';


test('работа кнопки Редактировать MobileClient', ()=>{
    let mode = 1;
    let client = {id:101, fio:{last_name: "Иванов", name: "Иван", middle_name: "Иванович"}, balance:200, status: "active"};

    const component = renderer.create(
        <MobileClient mode={mode} info={client}/>
    );
    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
    const buttonElem = component.root.find( el => el.props.className =='inputEdit');
    buttonElem.props.onClick();
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
    buttonElem.props.onClick();
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
});

test('работа кнопки Удалить MobileClient', ()=>{
    let mode = 1;
    let client = {id:101, fio:{last_name: "Иванов", name: "Иван", middle_name: "Иванович"}, balance:200, status: "active"};
    const component = renderer.create(
        <MobileClient mode={mode} info={client}/>
    );
    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
    const buttonElem = component.root.find( el => el.props.className=='inputDelete');
    buttonElem.props.onClick();
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
    buttonElem.props.onClick();
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
});

