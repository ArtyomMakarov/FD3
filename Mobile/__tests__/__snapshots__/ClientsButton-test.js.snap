// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа кнопки Редактировать MobileClient 1`] = `
<tr
  className="MobileClient"
>
  <td
    className="cell"
  >
    Иванов
  </td>
  <td
    className="cell"
  >
    Иван
  </td>
  <td
    className="cell"
  >
    Иванович
  </td>
  <td
    className="cell"
  >
    200
  </td>
  <td
    className="cell"
    style={
      Object {
        "background": "green",
      }
    }
  >
    active
  </td>
  <td
    className="cell"
  >
    <input
      className="inputEdit"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td
    className="cell"
  >
    <input
      className="inputDelete"
      disabled={true}
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
  </td>
</tr>
`;

exports[`работа кнопки Редактировать MobileClient 2`] = `
<tr
  className="MobileClient"
>
  <td
    className="cell"
  >
    Иванов
  </td>
  <td
    className="cell"
  >
    Иван
  </td>
  <td
    className="cell"
  >
    Иванович
  </td>
  <td
    className="cell"
  >
    200
  </td>
  <td
    className="cell"
    style={
      Object {
        "background": "green",
      }
    }
  >
    active
  </td>
  <td
    className="cell"
  >
    <input
      className="inputEdit"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td
    className="cell"
  >
    <input
      className="inputDelete"
      disabled={true}
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
  </td>
</tr>
`;

exports[`работа кнопки Редактировать MobileClient 3`] = `
<tr
  className="MobileClient"
>
  <td
    className="cell"
  >
    Иванов
  </td>
  <td
    className="cell"
  >
    Иван
  </td>
  <td
    className="cell"
  >
    Иванович
  </td>
  <td
    className="cell"
  >
    200
  </td>
  <td
    className="cell"
    style={
      Object {
        "background": "green",
      }
    }
  >
    active
  </td>
  <td
    className="cell"
  >
    <input
      className="inputEdit"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td
    className="cell"
  >
    <input
      className="inputDelete"
      disabled={true}
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
  </td>
</tr>
`;

exports[`работа кнопки Удалить MobileClient 1`] = `
<tr
  className="MobileClient"
>
  <td
    className="cell"
  >
    Иванов
  </td>
  <td
    className="cell"
  >
    Иван
  </td>
  <td
    className="cell"
  >
    Иванович
  </td>
  <td
    className="cell"
  >
    200
  </td>
  <td
    className="cell"
    style={
      Object {
        "background": "green",
      }
    }
  >
    active
  </td>
  <td
    className="cell"
  >
    <input
      className="inputEdit"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td
    className="cell"
  >
    <input
      className="inputDelete"
      disabled={true}
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
  </td>
</tr>
`;

exports[`работа кнопки Удалить MobileClient 2`] = `
<tr
  className="MobileClient"
>
  <td
    className="cell"
  >
    Иванов
  </td>
  <td
    className="cell"
  >
    Иван
  </td>
  <td
    className="cell"
  >
    Иванович
  </td>
  <td
    className="cell"
  >
    200
  </td>
  <td
    className="cell"
    style={
      Object {
        "background": "green",
      }
    }
  >
    active
  </td>
  <td
    className="cell"
  >
    <input
      className="inputEdit"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td
    className="cell"
  >
    <input
      className="inputDelete"
      disabled={true}
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
  </td>
</tr>
`;

exports[`работа кнопки Удалить MobileClient 3`] = `
<tr
  className="MobileClient"
>
  <td
    className="cell"
  >
    Иванов
  </td>
  <td
    className="cell"
  >
    Иван
  </td>
  <td
    className="cell"
  >
    Иванович
  </td>
  <td
    className="cell"
  >
    200
  </td>
  <td
    className="cell"
    style={
      Object {
        "background": "green",
      }
    }
  >
    active
  </td>
  <td
    className="cell"
  >
    <input
      className="inputEdit"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td
    className="cell"
  >
    <input
      className="inputDelete"
      disabled={true}
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
  </td>
</tr>
`;
